lib/Controller/LogoutController.php

```diff
@@ -0,0 +1,202 @@
+            }
+        }
+
+        // Set indication for OIDC initiated logout back to false, so that the logoutHandler() method does not
+        // run for other logout initiated actions, like (currently) re-authentication...
+        $this->sessionService->setIsOidcInitiatedLogout(false);
+
+        return $this->resolveResponse($logoutRequest, $wasLogoutActionCalled);
+    }
+
+    /**
+     * Logout handler function registered using Session::registerLogoutHandler() during authn.
+     * @throws Exception
+     */
+    public static function logoutHandler(): void
+    {
+        $session = Session::getSessionFromRequest();
+
+        // Only run this handler if logout was initiated using OIDC protocol. This is important since this
+        // logout handler will (currently) also be called in re-authentication cases.
+        // https://groups.google.com/g/simplesamlphp/c/-uhiVE8TaF4
+        if (! SessionService::getIsOidcInitiatedLogoutForSession($session)) {
+            return;
+        }
+
+        $relyingPartyAssociations = SessionService::getRelyingPartyAssociationsForSession($session);
+        SessionService::clearRelyingPartyAssociationsForSession($session);
+
+        // Check for session logout tickets. If there are any, it means that the logout was initiated using OIDC RP
+        // initiated flow for specific session (not current one).
+        $sessionLogoutTicketStore = SessionLogoutTicketStoreBuilder::getStaticInstance();
+        $sessionLogoutTickets = $sessionLogoutTicketStore->getAll();
+
+        if (! empty($sessionLogoutTickets)) {
+            foreach ($sessionLogoutTickets as $sessionLogoutTicket) {
+                $sid = $sessionLogoutTicket['sid'];
+                if ($sid === $session->getSessionId()) {
+                    continue;
+                }
+
+                try {
+                    if (($sessionLogoutTicketSession = Session::getSession($sid)) !== null) {
+                        $relyingPartyAssociations = array_merge(
+                            $relyingPartyAssociations,
+                            SessionService::getRelyingPartyAssociationsForSession($sessionLogoutTicketSession)
+                        );
+
+                        SessionService::clearRelyingPartyAssociationsForSession($sessionLogoutTicketSession);
+                    }
+                } catch (Throwable $exception) {
+                    LoggerService::getInstance()->warning(
+                        sprintf(
+                            'Session Ticket Logout: could not get session with ID %s, error: %s',
+                            $sid,
+                            $exception->getMessage()
+                        )
+                    );
+                }
+            }
+
+            $sessionLogoutTicketStore->deleteMultiple(array_map(fn($slt) => $slt['sid'], $sessionLogoutTickets));
+        }
+
+        (new BackChannelLogoutHandler())->handle($relyingPartyAssociations);
```

Comment:
Should this come from the Container/constructor injection?

Reply:
Check comment.